ATM/atm.py

```diff
@@ -0,0 +1,202 @@
+        for pw, balance in temp_dict.items():
+            if pw == password:
+                print('OK, we proceed ')
+                return True
+            else:
+                print('Wrong password, you have {} more tries'
+                      .format(times - 1))
+                times -= 1
+                if times == 0:
+                    return False
+
+
+def who_are_you(dictionary):
+    """
+    Ask from the user to enter his ID.
+    If the ID exist in the dictionary (in the atm.txt) so return the ID.
+    If the ID is -1, return also the ID.
+    If the ID doesn't exist in the dictionary, keep asking for an different ID.
+    :param dictionary:
+    :return: consumer_id (that exist in the file, or -1)
+    """
+    while True:
+        consumer_id = input('Please enter you ID: ')
+        if (consumer_id in dictionary) or (consumer_id == '-1'):
+            return consumer_id
+        else:
+            print('Sorry, you are not in the system, please try again! ')
+
+
+def read_file(file_name):
+    """
+    Read from the file (atm.txt) line by line
+    and create dictionary: {ID: {password: balance}}
+    I used try and except. If the file doesn't exist,
+    the program won't collapse
+    :param file_name:
+    :return: dictionary
+    """
+    try:
+        dictionary = {}
+        with open(file_name, 'r') as atm_file:
+            for line in atm_file:
+                my_list = line.split()
+                dictionary[my_list[ID]] = \
+                    {my_list[PASSWORD]: int(my_list[BALANCE])}
+        return dictionary
+    except Exception as error:
+        print(error)
+
+
+def main():
+    """
+    1) call read_file(file_name) to create dictionary
+    2) call who_are_you(dictionary) to recognize the user
+    3) call enter_password(dictionary, consumer_id)
+        a) if the password is correct in 3 tries then choose an action
+        b) else: Message about it, and go to step 2)
+    :return:
+    """
+    file_name = sys.argv[FILE_NAME]
+    dictionary = read_file(file_name)
+    while True:
+        consumer_id = who_are_you(dictionary)
+        if consumer_id == '-1':
+            print('Turning off the ATM...')
+            break
+        if enter_password(dictionary, consumer_id) is True:
+            # if the password is correct
+            while True:
+                action = input('Press:\n1 -- see balance\n' +
+                               '2 -- cash withdraw\n' +
+                               '3-- cash deposit\n' +
+                               '4 -- change password\n' +
+                               'STOP --  if you are done\n')
+                if action == '1':
+                    balance = check_the_balance(dictionary, consumer_id)
+                    print('Your balance is: {}'.format(balance))
+
+                elif action == '2':
+                    dictionary = cash_withdrawal(dictionary, consumer_id)
+                    update_file(file_name, dictionary)
```

Comment:
this pattern also called switch-case. It's where based on some conditions you apply some actions
In Python, you can do the following:
```python
OPTIONS = {
    "1": check_the_balance,
    "2": cash_withdrawal,
    "3": ...
}
```
then in this function you can do the following:
```python
action = input("Press:...")
action_function = ACTIONS[action]
action_function(dictionary, consumer_id)
```

This will make the relevant function to be called, as you do the same things in all functions (running the function, updating the DB, etc.)
You might need to change the `check_the_balance` function to just print the balance by itself